etl/steps/data/garden/agriculture/2023-04-21/uk_long_term_yields.py

```diff
@@ -0,0 +1,124 @@
+"""Load historical data on UK yields and combine it with the latest FAOSTAT data."""
+
+import pandas as pd
+from owid.catalog import Dataset, Table
+from owid.datautils.dataframes import combine_two_overlapping_dataframes
+from structlog import get_logger
+
+from etl.helpers import PathFinder, create_dataset
+
+log = get_logger()
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+# Element code for "Yield".
+ELEMENT_CODE_FOR_YIELD = "005419"
+
+# Item codes for required items.
+ITEM_CODE_FOR_WHEAT = "00000015"
+ITEM_CODE_FOR_BARLEY = "00000044"
+ITEM_CODE_FOR_OATS = "00000075"
+ITEM_CODE_FOR_POTATOES = "00000116"
+ITEM_CODE_FOR_PULSES = "00001726"
+ITEM_CODE_FOR_RYE = "00000071"
+ITEM_CODE_FOR_SUGAR_BEET = "00000157"
+ITEM_CODES = [
+    ITEM_CODE_FOR_WHEAT,
+    ITEM_CODE_FOR_BARLEY,
+    ITEM_CODE_FOR_OATS,
+    ITEM_CODE_FOR_POTATOES,
+    ITEM_CODE_FOR_PULSES,
+    ITEM_CODE_FOR_RYE,
+    ITEM_CODE_FOR_SUGAR_BEET,
+]
+
+
+def run(dest_dir: str) -> None:
+    log.info("uk_long_term_yields.start")
+
+    #
+    # Load inputs.
+    #
+    # Load UK long-term yields data from Broadberry et al. (2015).
+    ds_broadberry: Dataset = paths.load_dependency("broadberry_et_al_2015")
+
+    # Read main table from dataset.
+    tb_broadberry = ds_broadberry["broadberry_et_al_2015"]
+
+    # Create a convenient dataframe.
+    df_broadberry = pd.DataFrame(tb_broadberry).reset_index()
+
+    # Load UK long-term yields data from Broadberry et al. (2015).
+    ds_brassley: Dataset = paths.load_dependency("brassley_2000")
+
+    # Read main table from dataset.
+    tb_brassley = ds_brassley["brassley_2000"]
+
+    # Create a convenient dataframe.
+    df_brassley = pd.DataFrame(tb_brassley).reset_index()
+
+    # Load faostat data on crop and livestock production.
+    ds_qcl: Dataset = paths.load_dependency("faostat_qcl")
+
+    # Read main table from dataset.
+    tb_qcl = ds_qcl["faostat_qcl"]
+
+    # Create a convenient dataframe.
+    df_qcl = pd.DataFrame(tb_qcl).reset_index()
+
+    #
+    # Process data.
+    #
+    # Select required country, element and items.
+    df_qcl = df_qcl[
+        (df_qcl["country"] == "United Kingdom")
+        & (df_qcl["element_code"] == ELEMENT_CODE_FOR_YIELD)
+        & (df_qcl["item_code"].isin(ITEM_CODES))
+    ].reset_index(drop=True)
+
+    # Sanity check.
+    error = "Units for yield have changed."
+    assert list(df_qcl["unit"].unique()) == ["tonnes per hectare"], error
```

Comment:
You could try to make this check a bit more informative:

```python
units_expected = {"tonnes per hectare"}
units_present = set(df_qcl["unit"])
# Check if units in data present unknown units
units_unexpected = units_present.difference(units_expected)
assert not units_unexpected, f"Unexpected units: {units_unexpected}"
# Check if there are units in data missing
units_missing = units_expected.difference(units_present)
assert not units_unexpected, f"Missing units: {units_missing}"
```

Reply:
The most likely scenario for this check to fail is that faostat (or us) change the name of the units, without changing the actual units (e.g. "tonnes/hectare"). So I think the current check is fine.

Reply:
Agree, but I still think that a more informative message would be helpful here, specially if the message is not read by the author of the code (i.e. you)